src/key-value-pairs/internal.tsx

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import clsx from 'clsx';
+import styles from './styles.css.js';
+import { KeyValuePairsProps } from './interfaces';
+import { InfoLinkLabelContext } from '../internal/context/info-link-label-context';
+import ColumnLayout from '../column-layout/internal';
+import Box from '../box/internal';
+
+export { KeyValuePairsProps };
+
+const InternalKeyValuePairs = React.forwardRef(
+  ({ columns, className, ...rest }: KeyValuePairsProps, ref: React.Ref<HTMLDivElement>) => {
+    return (
+      <div {...rest} className={clsx(styles['key-value-pairs'], className)} ref={ref}>
+        {!!columns?.length && (
+          <ColumnLayout columns={Math.min(columns.length, 4)} variant="text-grid">
+            {columns.map((column, columnIndex) => (
+              <div className={styles.column} key={columnIndex}>
+                {column.title && (
+                  <Box variant="h3" padding="n">
+                    {column.title}
+                  </Box>
+                )}
+                <dl className={styles.list}>
+                  {column.items.map((item, itemIndex) => (
+                    <div key={columnIndex + itemIndex} className={styles.item}>
```

Comment:
Why was `key={itemIndex}` not enough here?

Reply:
Because there might be multiple columns, so `itemIndex` won't be unique

Reply:
The index need to be unique within a single parent, `<dl>` tag in this case. Other columns do not matter

Reply:
updated